### PR TITLE
improvement(k8s-eks): use EBS gp3 disks

### PR DESCRIPTION
**Pull request by vponomaryov**
Make 2 following main changes to switch all the EKS volume usages from the `gp2` to the `gp3` type:
- Explicitly specify the disk type for all the K8S nodes created as part of the K8S cluster
- Configure the `gp3` storage class for K8S PVC.

### Testing

- https://jenkins.scylladb.com/job/scylla-staging/job/valerii/job/vp-functional-eks/64

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

File: sdcm/cluster_k8s/eks.py [hunk 1]
```diff
@@ -214,29 +214,25 @@ def launch_template_name(self) -> str:
...
-            block_devices.append(root_disk_def)
+            root_ebs_block_device_params["VolumeSize"] = self.disk_size
+        root_disk_def = LaunchTemplateBlockDeviceMappingRequestTypeDef(
+            DeviceName="/dev/xvda",
+            Ebs=LaunchTemplateEbsBlockDeviceRequestTypeDef(**root_ebs_block_device_params))
         launch_template = RequestLaunchTemplateDataTypeDef(
             KeyName=self.ssh_key_pair_name,
             EbsOptimized=False,
-            BlockDeviceMappings=block_devices,
-            NetworkInterfaces=[{
```
Comment (fruch) [inline sdcm/cluster_k8s/eks.py:236]: This isn't needed ?

Comment (vponomaryov) [reply sdcm/cluster_k8s/eks.py:236]: It is prohibited here, because security groups are specified out of the launch-template already.

Comment (fruch) [reply sdcm/cluster_k8s/eks.py:236]: So before this change we weren't using the template at all ?

Do we clean those templates ?

**Review by soyacz — APPROVED** [review]
LGTM

File: sdcm/cluster_k8s/eks.py (hunk 1 above)
Comment (vponomaryov) [reply sdcm/cluster_k8s/eks.py:236]: Yes, we were not using `launch templates` before.
No, we do not clean those templates.

Comment (fruch) [reply sdcm/cluster_k8s/eks.py:236]: we should add the needed code for cleaning them up, in a followup PR

**Pull request merged**